src/Groups/GAPGroups.jl

```diff
@@ -0,0 +1,617 @@
+
+"""
+```
+    normalizer(G::Group, H::Group)
+    normaliser(G::Group, H::Group)
+```
+returns `N,f`, where `N` is the normalizer of `H` in `G` and `f` is the embedding morphism of `N` into `G`.
+"""
+normalizer(G::T, H::T) where T<:Group = _as_subgroup(GAP.Globals.Normalizer(G.X,H.X),G)
+
+"""
+```
+    normalizer(G::Group, x::GAPGroupElem)
+    normaliser(G::Group, x::GAPGroupElem)
+```
+returns `N,f`, where `N` is the normalizer of <`x`> in `G` and `f` is the embedding morphism of `N` into `G`.
+"""
+normalizer(G::Group, x::GAPGroupElem) = _as_subgroup(GAP.Globals.Normalizer(G.X,x.X),G)
+
+normaliser = normalizer
+
+"""
+    core(G::Group, H::Group)
+returns `C,f`, where `C` is the normal core of `H` in `G`, and `f` is the embedding morphism of `C` into `G`.
+"""
+core(G::T, H::T) where T<:Group = _as_subgroup(GAP.Globals.Core(G.X,H.X),G)
+
+# normal_closure(G::T, H::T) where T<:Group = T(GAP.Globals.NormalClosure(G.X,H.X))
+# we don't know how G,H embeds into N_C(G,H) 
+
+"""
+    pcore(G::Group, p::Int64)
+return `C,f`, where `C` is the `p`-core (i.e. the largest normal `p`-subgroup) of `G` and `f` is the embedding morphism of `C` into `G`.
+"""
+function pcore(G::Group, p::Int64)
+   if !GAP.Globals.IsPrime(p)
+      throw(ArgumentError("p is not a prime"))
+   end
+   return _as_subgroup(GAP.Globals.PCore(G.X,p),G)
+end
+
+
+
+################################################################################
+#
+# Specific Subgroups
+#
+################################################################################
+
+# commutator_subgroup(G::T, H::T) where T<:Group = T(GAP.Globals.CommutatorSubgroup(G.X,H.X))
+# we don't know how G,H embed into [G,H]
+
+fitting_subgroup(G::Group) = _as_subgroup(GAP.Globals.FittingSubgroup(G.X),G)
+
+frattini_subgroup(G::Group) = _as_subgroup(GAP.Globals.FrattiniSubgroup(G.X),G)
+
+radical_subgroup(G::Group) = _as_subgroup(GAP.Globals.RadicalGroup(G.X),G)
+
+socle(G::Group) = _as_subgroup(GAP.Globals.Socle(G.X),G)
+
+
+################################################################################
+#
+# Sylow & Hall Subgroups
+#
+################################################################################
+
+"""
+    sylow_subgroup(G::Group, p::Int64)
+return a Sylow `p`-subgroup of `G`.
+"""
+function sylow_subgroup(G::Group, p::Int64)
+   if !GAP.Globals.IsPrime(p)
+      throw(ArgumentError("p is not a prime"))
+   end
+   return _as_subgroup(GAP.Globals.SylowSubgroup(G.X,p),G)
+end
+
+"""
+    hall_subgroup(G::Group, P::Array{Int64})
+return a Hall `P`-subgroup of `G`. It works only if `G` is solvable.
+"""
+function hall_subgroup(G::Group, P::Array{Int64})
+   noprime=false
+   for p in P
+      if !GAP.Globals.IsPrime(p)
+         noprime=true
+         break
+      end
+   end
+   if length(P) != length(Set(P)) noprime=true end        # avoid repetitions in P
+   if noprime throw(ArgumentError("The integers must be prime")) end
```

Comment:
here the error is misleading: everything in `P` could be prime, but not unique

Reply:
Corrected. Now repetitions are allowed.

Reply:
But GAP doesn't allow repetitions, so now you'll get a GAP error instead. I recommend that when you tweak argument validation code like this, you always write tests covering it in parallel. So here in particular, there should be tests that call `hall_group` with e.g.
- an empty list
- one containing duplicates, e.g. `[2,2]`
- non-primes, e.g. [4]
- a group that is not solvable (GAP will return `fail` then -- are you dealing right with that? Might want to throw an exception instead?)

Reply:
I don't get a GAP error because the first action the function does is eliminate the repetition with the instruction `P = unique (P)`.
Hence, write `hall_subgroup (G, [2])` or `hall_subgroup (G, [2,2])` gets the same result. Do you prefer I do not allow repetitions in `P` and throw an error when there are repeated numbers, as in GAP?